src/nn/Operation.ts

```diff
@@ -0,0 +1,37 @@
+import { Output } from "./Output";
+import { Operand } from "./Operand";
+import { Constant } from "./Constant";
+import { Input } from "./Input";
+import { ExecutionContext } from "./ExecutionContext";
+import { assert } from "./utils";
+
+import * as tf from '@tensorflow/tfjs-core'
+
+export abstract class Operation {
+  inputs: Array<Operand> = [];
+  outputs: Array<Output> = [];
+
+  constructor(inputs: Array<Operand>) {
+    assert(inputs.every(input => input instanceof Operand), 'The inputs parameter is invalid.');
+    this.inputs = inputs;
+    this.outputs.push(new Output(this));
+  }
+
+  get output(): Output {
```

Comment:
there are ops generate multiple outputs

Reply:
The `get output()` is for first output. The `outputs` property itself is public. For a better implementation, I can add a `get outputs()`.

Reply:
I've added `get outputs()` in new commit. Please take a look.